### PR TITLE
feat: update basic-typescript example

### PR DESCRIPTION
**Pull request by apledger**

### CODE REVIEW

File: examples/basic-typescript/.env [hunk 1]
```diff
@@ -1 +1 @@
-MAKESWIFT_SITE_API_KEY=
+MAKESWIFT_SITE_API_KEY=a86621c2-5154-4c39-84f7-bbe2d3f9b48d
```
Comment (migueloller) [inline examples/basic-typescript/.env:1]: Hmm, I think you just leaked the API key for this site

File: examples/basic-typescript/pages/_app.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+const inter = Inter({
+  subsets: ['latin'],
+  variable: '--font-inter',
+})
+
+function MyApp({ Component, pageProps }: AppProps) {
+  useEffect(() => {
+    document.body.classList.add(inter.variable)
+  }, [])
```
Comment (migueloller) [inline examples/basic-typescript/pages/_app.tsx:17]: Why do we need to add the class to the body here? Won't this cause FOUC?

File: examples/basic-typescript/pnpm-lock.yaml
Comment (migueloller) [inline examples/basic-typescript/pnpm-lock.yaml:1]: Have you verified that changing from Yarn to pnpm will work with our CLI?

File: examples/basic-typescript/tsconfig.json [hunk 3]
```diff
@@ -13,7 +13,14 @@
     "resolveJsonModule": true,
     "isolatedModules": true,
     "jsx": "preserve",
-    "incremental": true
+    "incremental": true,
+    "baseUrl": ".",
+    "paths": {
+      "@/components/*": ["components/*"],
+      "@/generated/*": ["generated/*"],
```
Comment (migueloller) [inline examples/basic-typescript/tsconfig.json:20]: We don't seem to have a `generated` folder in this repo. I'm guessing this was copy/pasted from another repo?

**Review by migueloller — COMMENTED** [review]
Looks good 👍🏻

Most important thing is to make sure that things still work well with the CLI and Vercel deploy button. Have you tested those out yet?

File: examples/basic-typescript/pages/_app.tsx (hunk 2 above)
Comment (apledger) [reply examples/basic-typescript/pages/_app.tsx:17]: This is just for modals and anything that is portaled out of the `<main>` element.

File: examples/basic-typescript/pnpm-lock.yaml
Comment (apledger) [reply examples/basic-typescript/pnpm-lock.yaml:1]: What's the best way to test this?

File: examples/basic-typescript/tsconfig.json (hunk 3 above)
Comment (apledger) [reply examples/basic-typescript/tsconfig.json:20]: Will update.

Comment (apledger) [reply examples/basic-typescript/tsconfig.json]: Actually this is where icons are generated from the included `icons` command.

File: examples/basic-typescript/pages/_app.tsx (hunk 2 above)
Comment (migueloller) [reply examples/basic-typescript/pages/_app.tsx:17]: Got it. Can this not be added to the `body` in `_document`, then? It's how I see `create-next-app` do it for the App Router, it puts it in the root layout's `body` element.

Comment (migueloller) [reply examples/basic-typescript/pages/_app.tsx:17]: Or maybe `next/font` can't be used in `_document`... They don't seem to do it with `create-next-app` when using the pages router

File: examples/basic-typescript/pnpm-lock.yaml
Comment (migueloller) [reply examples/basic-typescript/pnpm-lock.yaml:1]: I vaguely remember @alexhwoods setting up some environment variables to be able to use local versions of the app as well as draft versions of changes by providing a full GitHub URL for example names. @alexhwoods, any pointers here?

File: examples/basic-typescript/tsconfig.json (hunk 3 above)
Comment (migueloller) [reply examples/basic-typescript/tsconfig.json:19]: Ah, got it, I missed that. Is that `icons` script being used in this example?

Comment (migueloller) [reply examples/basic-typescript/tsconfig.json:19]: We don't seem to have any icons under `public/icons` so maybe we just want to clean that stuff up?

File: examples/basic-typescript/.env [hunk 4]
```diff
@@ -1 +1 @@
-MAKESWIFT_SITE_API_KEY=
+MAKESWIFT_SITE_API_KEY=
```
Comment (migueloller) [inline examples/basic-typescript/.env:1]: Missing a newline at the end of the file

**Review by joshuawootonn — APPROVED** [review]

**Pull request merged**